zebra/zapi_msg.c

```diff
@@ -948,6 +948,7 @@ int zsend_pw_update(struct zserv *client, struct zebra_pw *pw)
 	stream_write(s, pw->ifname, IF_NAMESIZE);
 	stream_putl(s, pw->ifindex);
 	stream_putl(s, pw->status);
+	stream_putl(s, pw->pwstatus);
```

Comment:
I think it would be better to extend `pw->status` instead of adding a new variable to convey additional information.

You could ditch `PW_STATUS_DOWN` and `PW_STATUS_UP` and replace them with the RFC4446 `PW_*` status flags you moved to _lib/pw.h_. Having to deal with a single type of pseudowire status would make things easier in my opinion.

What do you think?

Reply:
@rwestphal I think that is a good suggestion.  

I'll update the PR with this change.  Thanks, Karen

Reply:
I think it is a good suggestion too. On the other hand, the code above is in the pw manager of zebra (right?), which is also used by static pseudowires, while RFC4446  and the corresponding codes are used only in case of signaling the PW in LDP.

Reply:
Yes, that suggestion implies that both zebra and ldpd would need to be updated to use the new pseudowire status flags.

As an example, a check like `if (pw->status == PW_STATUS_UP)` would need to be updated to `if (pw->status == PW_FORWARDING)`.

Reply:
Ok. I have one comment here. 
**pw->status** seems to be naturally interpreted as a boolean ( non-zero meaning up ). In fact, PW_STATUS_UP is  defined as 1 and, in some places, I can read `if ( pw->status )`.
With your proposal, we would need to change all those occurrences by `(pw->status == PW_FORWARDING)`. That's fine, but PW_FORWARDING is defined as 0 (which is a bit counter-intuitive) ... and those values cannot be changed as are standard / IANA allocated, right?. 

Would something like renaming pw->pwstatus as pw->**ldp_status** (or similar, somehow stressing the fact that those are just the bits to announce to a peer in the case of LDP and if the status TLV is enabled ) make the code simpler and less prone to  issues in the future ?

Reply:
> Ok. I have one comment here.
> **pw->status** seems to be naturally interpreted as a boolean ( non-zero meaning up ). In fact, PW_STATUS_UP is defined as 1 and, in some places, I can read `if ( pw->status )`.
> With your proposal, we would need to change all those occurrences by `(pw->status == PW_FORWARDING)`. That's fine, but PW_FORWARDING is defined as 0 (which is a bit counter-intuitive) ... and those values cannot be changed as are standard / IANA allocated, right?.

Right. But I don't see changing all occurrences of `pw->status` as a big problem (there are only 10 of them considering both zebra and ldpd). Also, I think testing `if (pw->status)` without comparing it to any value isn't a good idea from a code readability perspective.

> Would something like renaming pw->pwstatus as pw->**ldp_status** (or similar, somehow stressing the fact that those are just the bits to announce to a peer in the case of LDP and if the status TLV is enabled ) make the code simpler and less prone to issues in the future ?

IMO that would only lead to more confusion. I don't see see how having two status fields would make things simpler. Also, the RFC4446 pseudowire status flags aren't LDP specific. That RFC is referenced by several other ones related to pseudowires. While only ldpd is making use of those flags now, that might change in the future (there's even an RFC specifying how to signal the status of static pseudowires).